controllers/discovery/initDiscoveryUserData.js

```diff
@@ -70,6 +70,10 @@ const InitDiscoveryUserData = async (req, res) => {
             A.karma_score,
             A.allow_anon_dm,
             A.only_received_dm_from_user_following,
+            ARRAY( select name from topics as tp
+              left join user_topics as utp on tp.topic_id = utp.topic_id
+              where utp.user_id = A.user_id limit 3
+            ) as community_info,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The query for `community_info` is using a `LIMIT 3` clause, which means it will only fetch the first three topics associated with the user. If there are more than three topics associated with the user, this could lead to incomplete data being returned. If the frontend requires all topics, you should consider removing the limit or increasing it to a number that ensures all relevant data is fetched.
```diff
- ARRAY( select name from topics as tp
-   left join user_topics as utp on tp.topic_id = utp.topic_id
-   where utp.user_id = A.user_id limit 3
- ) as community_info,
+ ARRAY( select name from topics as tp
+   left join user_topics as utp on tp.topic_id = utp.topic_id
+   where utp.user_id = A.user_id
+ ) as community_info,
```
If the limit of 3 is intentional and you only want to fetch the top 3 topics for some reason (like showing only the top 3 topics in the UI), then the code is fine as it is.


<!-- This is an auto-generated comment by OSS CodeRabbit -->